cleverhans/utils_pytorch.py

```diff
@@ -183,7 +183,8 @@ def optimize_linear(grad, eps, ord=np.inf):
 
     max_abs_grad, _ = torch.max(abs_grad.view(grad.size(0), -1), 1)
     max_mask = abs_grad.eq(max_abs_grad.view(ori_shape)).to(torch.float)
-    num_ties = torch.sum(max_mask, red_ind, keepdim=True)
```

Comment:
@npapernot I know this has already been merged but out of curiosity, could you please let me know what was the test case that revealed a bug in the original implementation? I've been testing with a few hand-crafted grad matrices but couldn't find a bug in the original implementation, so I guess I must have missed some corner cases. Thanks.